core/schema/container.graphqls

```diff
@@ -138,7 +127,7 @@ type Container {
     # FIXME: this is the last case of an actual "verb" that cannot cleanly go away.
     #    This may actually be a good candidate for a mutation. To be discussed.
     "Publish this container as a new image"
-    publish(address: ContainerAddress!): ContainerAddress!
+    publish(address: String!): Boolean!
```

Comment:
I'm a bit worried about the change to `Boolean`, at least from a `dagger-classic` perspective. I'll be needing the full ref for the pushed image (including the image digest) in order to get full compatibility. I can also imagine that it could be important to have that for pipelines that want to push the container, then ensure that the same container ref is used further down the line for deployment.

I'm not sure what a better return type would be... but just want to name it before totally settle on a API.

Reply:
Oh good catch! This was actually busted before, then, because it would always just return the ref that was given, which felt a little pointless. I'll change it to String and add a test that it returns a fully qualified ref with the digest and everything. :+1:

Reply:
Latest changes re-introduce this and test that we get a digest back.

Reply:
Looks good, @vito!